api/NetUtils.go

```diff
@@ -0,0 +1,34 @@
+package api
+
+import (
+	"strconv"
+	"encoding/json"
+	"net/http"
+	"io/ioutil"
+	"errors"
+)
+
+type StringResult struct {
+	Result string
+	Err error
+}
+
+// getBodyJson takes an http.Response pointer and reads all of its data into a map
+func getBodyJson(response *http.Response) (map[string] string, error) {

```

Comment:
when i better understand the overall project I'll have better input. But I have a feeling there is a better way to represent the body than `map[string]string`. A lot of times people make `structs` with `json` tags and when you throw that into `json.Unmarshal` it works easy. Then you have your own custom representation that you can cater towards readability. ill find a link

Reply:
I think this is the right link. Its for `marshal` but works both ways
https://golang.org/pkg/encoding/json/#Marshal